runtime/commonMain/src/kotlinx/serialization/internal/GeneratedSerializer.kt

```diff
@@ -10,3 +10,17 @@ import kotlinx.serialization.*
 public interface GeneratedSerializer<T> : KSerializer<T> {
     fun childSerializers(): Array<KSerializer<*>>
 }
+
+/**
+ * An internal interface which compiler plugin may sometimes add as a superinterface for objects
+ * that can provide serializers (currently it is done for auto-generated companion objects of @Serializable classes with type parameters).
+ * This interface is used mainly to overcome lack of reflection on Kotlin/Native
+ * via internal mechanisms provided by its compiler (see @AssociatedObjectKey).
+ * Should not be used in any user code. Please use generated `.serializer(kSerializer1, kSerializer2, ...)`
+ * method on a companion or top-level `serializer(KType)` function.
+ */
+@InternalSerializationApi
+@Deprecated("Inserted into generated code and should not be used directly", level = DeprecationLevel.HIDDEN)
+public interface SerializerFactory {
```

Comment:
The outer file should no longer be named `GeneratedSerializer`.

Reply:
WDYT of `PluginHelperInterfaces.kt` ?

Reply:
I'd prefer something more expressive, without a `*Helper*`, e.g. `PluginInterfaces`/`RuntimeInterfaces`
Anyway, works for me.